Modules/Sources/WordPressUI/Views/Users/Views/UserDetailsView.swift

```diff
@@ -175,12 +193,124 @@ struct UserDetailsView: View {
+            "userdetail.alert.deleteUserConfirmationTitle",
+            value: "Are you sure?",
+            comment: "The title of the alert that appears when deleting a user"
+        )
+
+        static func deleteUserConfirmationMessage(username: String) -> String {
+            let format = NSLocalizedString(
+                "userdetail.alert.deleteUserConfirmationMessage",
+                value: "Are you sure you want to delete this user and attribute all content to %@?",
+                comment: "The message in the alert that appears when deleting a user. The first argument is the display name of the user to which content will be attributed"
+            )
+            return String(format: format, username)
+        }
+
+        static let deleteUserConfirmButtonTitle = NSLocalizedString(
+            "userdetail.alert.deleteUserConfirmButtonTitle",
+            value: "Yes, delete user",
+            comment: "The title of the confirmation button in the alert that appears when deleting a user"
+        )
     }
 }
 
-#Preview {
-    NavigationStack {
-        UserDetailsView(user: DisplayUser.MockUser, userProvider: MockUserProvider(), actionDispatcher: UserManagementActionDispatcher())
+private extension View {
+    @ViewBuilder
+    func deleteUser(in view: UserDetailsView) -> some View {
+        sheet(
+            isPresented: view.$presentUserPicker,
+            onDismiss: {
+                view.presentUserPicker = false
+            },
+            content: {
+                pickAnotherUser(in: view)
+            }
+        )
+        .alert(
+            UserDetailsView.Strings.deleteUserConfirmationTitle,
+            isPresented: view.$presentDeleteConfirmation,
+            presenting: view.selectedUser,
+            actions: { attribution in
+                Button(role: .destructive) {
+                    Task {
+                        view.deleteUserViewModel.didTapDeleteUser {
+                            view.dismissAction()
+                        }
+                    }
+                } label: {
+                    Text(UserDetailsView.Strings.deleteUserConfirmButtonTitle)
+                }
+            },
+            message: {
+                Text(UserDetailsView.Strings.deleteUserConfirmationMessage(username: $0.displayName))
+            }
+        )
+        .alert(
+            "Error",
+            isPresented: view.$presentDeleteUserError,
+            presenting: view.deleteUserViewModel.error,
+            actions: { _ in
+                Button("OK") {
+                    view.presentDeleteUserError = false
+                }
+            },
+            message: { error in
+                Text("An error occurred while deleting the user.")
+                // TODO: Use appropriate localized error message
+                Text(error.localizedDescription)
+            })
+    }
+
+    @ViewBuilder
+    func pickAnotherUser(in view: UserDetailsView) -> some View {
```

Comment:
Missing localization in this method in a few other places in this file.

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/23791/commits/fe92323da434a1188cdcfa9210915828b626f506